src/components/productDetail.ts

```diff
@@ -10,7 +10,7 @@ function productDetail(product: Product | undefined): string {
     <h2 class="nft--author">${product.author}</h2>
     <p class="nft--details">${product.description}</p>
     <h3 class="nft--price">$${product.price}</h3>
-    <button class="btn buy-now js-add-to-cart" data-product-id="${product.id}">Buy now</button>
```

Comment:
si hay javascript actions sobre el element, es buena practica dejarle el js- 

https://www.freecodecamp.org/news/css-naming-conventions-that-will-save-you-hours-of-debugging-35cea737d849/#1-use-js-class-names